gui/dialog/set_energy.py

```diff
@@ -0,0 +1,138 @@
+        self.monochromator_button.clicked.connect(self.set_monochromator_energy)
+
+        self.full_alignment_button = QtWidgets.QPushButton("Full Alignment")
+        self.full_alignment_button.setAutoDefault(False)
+        self.full_alignment_button.clicked.connect(self.set_full_alignment_energy)
+
+        self.close_button = QtWidgets.QPushButton("Close")
+        self.close_button.clicked.connect(self.close)
+        self.close_button.setAutoDefault(False)
+
+        layout.addWidget(self.monochromator_button, 3, 0)
+        layout.addWidget(self.full_alignment_button, 3, 1)
+        layout.addWidget(self.close_button, 4, 0, 1, 2)
+
+        self.hdcm.e.user_readback.subscribe(self.update_energy, run=True)
+
+        self.energy_changed_signal.connect(
+            lambda value: self.current_energy_value_label.setText(f"{value:.2f}")
+        )
+
+        self.setLayout(layout)
+        self.setModal(True)
+        self.show()
+
+    def update_energy(self, value, old_value, **kwargs):
+        self.energy_changed_signal.emit(value)
+
+    def check_value(self):
+        if abs(float(self.setpoint_edit.text()) - self.hdcm.e.user_readback.get()) > 10:
+            self.message.setText(
+                "Energy change is greater than 10 eV.\nMonochromator cannot be used for alignment"
+            )
+            self.monochromator_button.setDisabled(True)
+        else:
+            self.message.setText("Energy change less than 10 eV")
+            self.monochromator_button.setDisabled(False)
+
+        if float(self.setpoint_edit.text()) <= 10000:
+            self.message.setText(
+                f"{self.message.text()} \n Beam shape not automatically optimized \nfor energies between 5 keV and 10 keV"
+            )
+
+    def unmount_cold_dialog(self):
+        msg_box = QtWidgets.QMessageBox()
+        msg_box.setText("A sample is mounted. Unmount cold?")
+        msg_box.setStandardButtons(QtWidgets.QMessageBox.StandardButton.Ok | QtWidgets.QMessageBox.StandardButton.Cancel)  # type: ignore
+        msg_box.setDefaultButton(QtWidgets.QMessageBox.StandardButton.Ok)
+        return msg_box
+
+    def set_full_alignment_energy(self):
+        if self._parent.mountedPin_pv.get():
+            # If sample is mounted, ask user to unmount cold
+            response = self.unmount_cold_dialog().exec_()
+
+            if response == QtWidgets.QMessageBox.Ok:
+                self._parent.send_to_server("unmountCold")
+            else:
+                return
+
+        if self._parent.governorMessage.getEntry().text() not in [
+            "state SE",
+            "state BA",
+            "state BL",
+            "state XF",
+            "state SA",
+        ]:
+            self.message.setText("Governor not in a valid state, call staff!")
+            return
+        else:
+            self._parent.send_to_server("setGovState", ['SA'])
+
+        self._parent.send_to_server(f"set_energy", [float(self.setpoint_edit.text())])
```

Comment:
this needs to be added to the whitelist: `daq_main_common.functions`

Reply:
Added https://github.com/NSLS-II/lsdc/pull/369/commits/5ffca9cd40b49ed37178e8f2f0d00cc72a85cc1a